playwright/src/main/java/com/microsoft/playwright/junit/impl/OptionsExtension.java

```diff
@@ -0,0 +1,36 @@
+package com.microsoft.playwright.junit.impl;
+
+import com.microsoft.playwright.PlaywrightException;
+import com.microsoft.playwright.junit.Options;
+import com.microsoft.playwright.junit.OptionsFactory;
+import com.microsoft.playwright.junit.UsePlaywright;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import static com.microsoft.playwright.junit.impl.ExtensionUtils.getUsePlaywrightAnnotation;
+
+public class OptionsExtension implements AfterAllCallback {
+  private static final ThreadLocal<Options> threadLocalOptions = new ThreadLocal<>();
+
+  @Override
+  public void afterAll(ExtensionContext extensionContext) {
+    threadLocalOptions.remove();
+  }
+
+  static Options getOptions(ExtensionContext extensionContext) {
+    Options options = threadLocalOptions.get();
+    if (options != null) {
+      return options;
+    }
+
+    UsePlaywright usePlaywrightAnnotation = getUsePlaywrightAnnotation(extensionContext);
+    try {
+      OptionsFactory configFactory = usePlaywrightAnnotation.optionsFactory().newInstance();
```

Comment:
Perhaps we can avoid this indirection and allow passing options sub class as a parameter instead, so that it can be instantiated here? It'd require extending `Options` class every time instead of just setting properties on one and the same type but would eliminate the indirection added by the factory. It seems to be roughly the same amount of code on the client end with both approaches so I don't have strong preference. Any thoughts on this?

Reply:
makes sense, fixed.